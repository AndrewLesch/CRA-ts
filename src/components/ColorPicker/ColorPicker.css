.color-container {
  display: flex;
  margin-left: 5%;
}

.color-button {
  width: 38px;
  height: 38px;
  border-radius: 50%;
  margin: 10px 8px 0 0;
  border: none;
}

.color-button:hover {
  outline: 2px solid var(--border-color-color-picker);
  cursor: pointer;
}

.color-button--selected {
  outline: 2px solid var(--border-color-color-picker);
  cursor: pointer;
}
