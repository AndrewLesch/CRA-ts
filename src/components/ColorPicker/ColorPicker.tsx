import React from 'react';

import './ColorPicker.css';

type ColorPickerProps = {
  colors: Array<string>;
  currentColor: string;
  onChange(color: string): void;
};

const ColorPicker: React.FC<ColorPickerProps> = ({
  colors,
  currentColor,
  onChange,
}) => (
  <div className="color-container">
    {colors.map((color) => (
      <button
        type="button"
        key={color}
        onClick={() => onChange(color)}
        className={`color-button ${
          currentColor === color ? 'color-button--selected' : ''
        }`}
        style={{ backgroundColor: color }}
      />
    ))}
  </div>
);

export default ColorPicker;
