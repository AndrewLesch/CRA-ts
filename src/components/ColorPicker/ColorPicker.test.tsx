import React from 'react';
import Enzyme, { mount, ReactWrapper } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import ColorPicker from './ColorPicker';

Enzyme.configure({ adapter: new Adapter() });

describe('ColorPicker Test', () => {
  const colors: Array<string> = ['color1', 'color2', 'color3'];
  let pickedColor: string = 'color2';

  const onChange = (color: string): void => {
    pickedColor = color;
  };
  let component: ReactWrapper;

  beforeEach(() => {
    component = mount(
      <ColorPicker
        colors={colors}
        currentColor={colors[1]}
        onChange={onChange}
      />
    );
  });

  it('test mounting ColorPicker', () => {
    expect(component).toMatchSnapshot();
  });

  it('test presence of buttins', () => {
    const button: ReactWrapper = component.find('button');
    expect(button).toHaveLength(3);
  });

  it('first button click test', () => {
    const btn: ReactWrapper = component.find('.color-button').first();
    btn.simulate('click');
    expect(pickedColor).toEqual('color1');
  });
});
