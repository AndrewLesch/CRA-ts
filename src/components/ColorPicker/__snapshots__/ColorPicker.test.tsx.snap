// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker Test should match snapshot 1`] = `
<ColorPicker
  colors={
    Array [
      "#dfb6b6",
      "#c3dfb6",
      "#dfd4b6",
    ]
  }
  currentColor="#c3dfb6"
  onChange={[Function]}
>
  <div
    className="color-container"
  >
    <button
      className="color-button "
      key="#dfb6b6"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfb6b6",
        }
      }
      type="button"
    />
    <button
      className="color-button color-button--selected"
      key="#c3dfb6"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#c3dfb6",
        }
      }
      type="button"
    />
    <button
      className="color-button "
      key="#dfd4b6"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#dfd4b6",
        }
      }
      type="button"
    />
  </div>
</ColorPicker>
`;
